src/iotHubDirectMethodExplorer.ts

```diff
@@ -27,34 +29,62 @@ export class IotHubDirectMethodExplorer extends BaseExplorer {
             return;
         }
 
-        vscode.window.showInputBox({ prompt: `Enter [Method Name] sent to [${deviceItem.deviceId}]` }).then((methodName: string) => {
-            if (methodName !== undefined) {
-                vscode.window.showInputBox({ prompt: `Enter [Payload] sent to [${deviceItem.deviceId}]` }).then((payload: string) => {
-                    let methodParams = {
-                        methodName,
-                        payload,
-                        timeoutInSeconds: 10,
-                        connectTimeoutInSeconds: 10,
-                    };
-                    let serviceClient = ServiceClient.fromConnectionString(iotHubConnectionString);
-                    this._outputChannel.show();
-                    this.outputLine(Constants.IoTHubDirectMethodLabel, `Invokeing Direct Method [${methodName}] to [${deviceItem.deviceId}] ...`);
-                    serviceClient.open((error) => {
-                        if (error) {
-                            this.outputLine(Constants.IoTHubDirectMethodLabel, error.message);
-                        } else {
-                            serviceClient.invokeDeviceMethod(deviceItem.deviceId, methodParams, (err, result) => {
-                                if (err) {
-                                    this.outputLine(Constants.IoTHubDirectMethodLabel, `Failed to invoke Direct Method: ${err.message}`);
-                                } else {
-                                    this.outputLine(Constants.IoTHubDirectMethodLabel, `Response from [${deviceItem.deviceId}]:`);
-                                    this.outputLine(Constants.IoTHubDirectMethodLabel, JSON.stringify(result, null, 2));
-                                }
-                            });
-                        }
-                    });
-                });
+        this.invokeDirectMethod(iotHubConnectionString, deviceItem.deviceId);
+    }
+
+    public async invokeModuleDirectMethod(moduleItem: ModuleItem) {
+        let iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);
+        if (!iotHubConnectionString) {
+            return;
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIInvokeModuleMethodEvent);
+        this.invokeDirectMethod(iotHubConnectionString, moduleItem.deviceId, moduleItem.moduleId);
+    }
+
+    private invokeDirectMethod(iotHubConnectionString: string, deviceId: string, moduleId?: string) {
+        const target = moduleId ?  `[${deviceId}]-[${moduleId}]` : `[${deviceId}]`;

```

Comment:
how about use "/" to separate deviceId/moduleId what is consistent with the rest api

Reply:
Good suggestion!